bool reset_w_distance
---
bool result